RSA Documentation Improved in encipher_rsa and decipher_rsa

hannah-kari opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests . Please also
write a comment on that issue linking back to this pull request once it is
open. -->
Fixes #16181 

#### Brief description of what is fixed or changed
Documentation of RSA improved. Includes information regarding encrypting with private keys and deciphering with public keys.
Examples also edited to account for these changes.

#### Other comments


#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* crypto
   * fixed documentation and examples in encipher_rsa to include private key encryption
   * fixed documentation and examples in decipher_rsa to include public key decryption
<!-- END RELEASE NOTES -->

shivanikohlii committed:
removed unused code

shivanikohlii committed:
Merge pull request #3 from shivanikohlii/Issue-15584-Ascii-generates-unicode

Issue 15584 ascii generates unicode

hannah-kari committed:
RSA Documentation Improvements

shivanikohlii committed:
Merge branch 'master' of https://github.com/sympy/sympy

Merge master

hannah-kari committed:
Merge branch 'master' into Issue-16181

sylee957 commented:
You would have to remove the trailing whitespaces to make the test succeed.

And also, I think your PR contains some redundant changes from the merge commit.
I believe this process can be cleanly done by rebasing or cherry-picking.

oscargus commented on sympy/crypto/crypto.py:
```diff
@@ -1266,44 +1266,60 @@ def rsa_private_key(p, q, e):
     Examples
     ========
-
+	
```
Here are spaces that should be removed to pass the tests.

oscargus commented on sympy/crypto/crypto.py:
```diff
@@ -1266,44 +1266,60 @@ def rsa_private_key(p, q, e):
+    >>> decipher_rsa(new_msg, puk)
     12
-
+	
```
And here.

hannah-kari committed:
Update crypto.py

hannah-kari commented:
Thank you both so much for your feedback! I've hopefully fixed the issues with the trailing whitespace, but please let me know if there are any other issues that need to be addressed.

oscargus commented:
AssertionError: File contains tabs instead of spaces: /home/travis/build/sympy/sympy/sympy/crypto/crypto.py, line 1271.

So you must use spaces instead of tabs.

hannah-kari committed:
Update crypto.py

hannah-kari commented:
Thank you! I just updated the file again. I believe I removed all of the tabs and replaced them with spaces, but please let me know if there is anything else I need to fix.

sylee957 commented on sympy/crypto/crypto.py:
```diff
@@ -1266,44 +1266,57 @@ def rsa_private_key(p, q, e):
     ========
-
+    *Public Key Encryption*
+    
```
There should be no hidden space here, to pass the tests.
I'd recommend you to configure your editor , to trim spaces automatically.

sylee957 commented on sympy/crypto/crypto.py:
```diff
@@ -1266,44 +1266,57 @@ def rsa_private_key(p, q, e):
     >>> encipher_rsa(msg, puk)
     3
-
+    
```
And also this.

shivanikohlii committed:
Removed trailing whitespace

sylee957 commented on sympy/physics/vector/printing.py:
```diff
@@ -156,7 +156,6 @@ def _print_Derivative(self, der_expr):
             base += '_' + base_split[1]
         return base
 
-
```
Can you remove some redundant changes in this module?

shivanikohlii replied:
We don't know why this module is showing up. It is not really showing up when we do rebase. I'm not sure how to fix this.

sylee957 replied:
Well, I don't see any clear indication of rabasing in here. 
Because after rewriting the history, there will be a footprint here about force pushing

There is a tutorial of interactive rebasing
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History
And you may do interactive rebasing to the upstream master, then you will have some option to drop some redundant commits.
See if it works.

sylee957 commented on sympy/crypto/crypto.py:
```diff
@@ -1266,44 +1266,55 @@ def rsa_private_key(p, q, e):
-    >>> decipher_rsa(msg, prk)
+    >>> puk = rsa_public_key(p, q, e)
+    >>> msg = 12
+    >>> new_msg = encipher_rsa(msg, prk)
```
The test may be failing because some functions like these are not imported.

Besides, I think the examples can be formatted as `encipher_rsa`
There can be two sections like `Public Key Decryption` and `Private Key Decryption`,
and add corresponding processes, matching examples above

sylee957 commented:
Apologize that I had not taken much attention to guiding this PR for a while,
but I've decided to merge some documentation changes here with my project in #17371. 
If you have some time and knowledge for this subject, please take a look.
